### PR TITLE
new package: sfeed -- a scriptable feed aggregator

### PR DESCRIPTION
**Pull request by flosnvjx**
*tldr*: this package consists of a shell scriptable feed aggregator, a RSS/Atom-to-TSV parser, and a curses-based feed entires reader.

Project is actively maintained and written in portable C and POSIX shell, listed on [suckless.org recommends](http://suckless.org/rocks).

------

`sfeed(1)` is a RSS/Atom feed parsing utility written in C that aimed to be used in a scriptable feed aggregating pipeline (or newsreading workflow, for a more generally speaking) on UNIX.

`sfeed_update(1)` is the POSIX shell script aimed to be placed in crontab for periodical updating of feeds, as the "aggregator".

**NOTE:** Crontab entries is not defined in `build.sh`.  On Termux things in crontab usually read/write to `$PREFIX/var`,
I'd leave this simple task to users for more flexibility (i.e. I may want to write to homedir or even adaptive storage instead, with different params.)

(See [README](https://codemadness.org/git/sfeed/file/README.html#l176), [manpages](https://man.voidlinux.org/sfeed.1) and `sfeed_update(1)` source code to get a feel of overall workflow.)

Once the feeds are updated, it will be totally up to user that how the entires should be handled, for examples:

 - send/sync feed entries to a mailbox using `sendmail(1)`/`isync(1)`, then read entries in Gmail App.
 - read gathered feed entries via TUI using `sfeed_curses(1)`
 - `sfeed_html(1)`, then using something like miniserve to read entries in Web browser.

### CODE REVIEW

File: packages/sfeed/build.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+TERMUX_PKG_MAINTAINER='@termux'
+## ref: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=sfeed
+TERMUX_PKG_VERSION='1.6'
+TERMUX_PKG_SRCURL='https://codemadness.org/releases/sfeed/sfeed-'$TERMUX_PKG_VERSION'.tar.gz'
+TERMUX_PKG_SHA256=bfd6d24ce98619726aa411a7a0d969806ad08a73c9adc3a8d04508e00eea6aea
+TERMUX_PKG_DEPENDS='ncurses'
+TERMUX_PKG_RECOMMENDS='curl'
+TERMUX_PKG_SUGGESTS='lynx, termux-tools'
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_MAKE_INSTALL_TARGET="PREFIX=$TERMUX_PREFIX MANPREFIX=$TERMUX_PREFIX/share/man install"
```
Comment (Grimler91) [inline packages/sfeed/build.sh:13]: TERMUX_PKG_EXTRA_MAKE_ARGS would be more suitable for these variables (since they aren't a make install target). Looks like the Makefile in any case needs a patch to make it use system $CPPFLAGS, so you could patch it so that it respects PREFIX at the same time, i.e. having `PREFIX ?= /usr/local`, and thereby get rid of this line

**Review by Grimler91 — CHANGES_REQUESTED** [review]
Thanks for the PR, see comments

File: packages/sfeed/build.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+TERMUX_PKG_HOMEPAGE='https://codemadness.org/sfeed-simple-feed-parser.html'
+TERMUX_PKG_DESCRIPTION='Shell-script/crontab oriented feed aggregator and parser utility (curses-based reader included)'
+TERMUX_PKG_LICENSE='ISC'
+TERMUX_PKG_MAINTAINER='@termux'
+## ref: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=sfeed
```
[the remark this answers is not in the document]
Comment (flosnvjx) [reply packages/sfeed/build.sh]: Nothing is special in Makefile/spec etc, it is just my bad habbit (old days when I writin pkgbuild on aur) and that line could indeed be deleted.

File: packages/sfeed/build.sh (hunk 1 above)
Comment (flosnvjx) [reply packages/sfeed/build.sh]: Thanks for review and comments!  Modified the commit according your comments.  It is may fault that send a PR w/o testing on Github CI first (the local build-package.sh does success tho).  Now it should works.

**Comment by 2096779623** [comment]
```bash
~/.sfeed/feeds $ sfeed_update
Configuration file "/data/data/com.termux/files/home/.sfeed/sfeedrc" cannot be read.
See sfeedrc.example for an example.
~/.sfeed/feeds $ cat tests
feeds() {
        feed "https://termux.dev/feed.xml"
}
```

When I execute `sfeed_web https://termux.dev/feed.xml`, no results are returned and the process is not ended, `$HOME/.sfeed/` needs to be created manually.

**Comment by flosnvjx** [comment]
> ```shell
> ~/.sfeed/feeds $ sfeed_update
> Configuration file "/data/data/com.termux/files/home/.sfeed/sfeedrc" cannot be read.
> See sfeedrc.example for an example.
> ~/.sfeed/feeds $ cat tests

`~/.sfeed/feeds` does need to exist and contain some entries.  Or do we need to document this somewhere?

> ```shell
> ~/.sfeed/feeds $ sfeed_update
> Configuration file "/data/data/com.termux/files/home/.sfeed/sfeedrc" cannot be read.
> See sfeedrc.example for an example.
> ~/.sfeed/feeds $ cat tests
> # https://man.voidlinux.org/sfeedrc.5
> feeds() {
>         feed "https://termux.dev/feed.xml"
> }
> ```
> 
> When I execute `sfeed_web https://termux.dev/feed.xml`, no results are returned and the process is not ended, `$HOME/.sfeed/` needs to be created manually.

The `sfeed_web` utility does not `curl`'ing by itself (which unlike `pandoc`), but read html page content from stdin instead.  It just like https://github.com/DIYgod/RSSHub-Radar but the user need to manually specify the base URL (if the feed URI is relative.) since it is not Web browser, `sfeed_web` have no knowledge about the URL of input HTML page on stdin.

**Comment by 2096779623** [comment]
> `~/.sfeed/feeds` does need to exist and contain some entries. Or do we need to document this somewhere?

It needs to be created manually in the documentation of void linux.

> The `sfeed_web` utility does not `curl`'ing by itself (which unlike `pandoc`), but read html page content from stdin instead. It just like https://github.com/DIYgod/RSSHub-Radar but the user need to manually specify the base URL (if the feed URI is relative.) since it is not Web browser, `sfeed_web` have no knowledge about the URL of input HTML page on stdin.

```bash
~ $ curl -s 'https://termux.dev/feed.xml' | sfeed_web
https://termux.dev/feed.xml     application/atom+xml
```

It looks like it's working fine now.

**Comment by Grimler91** [comment]
Thanks!
